src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -623,6 +623,36 @@ private function step_in_body() {
 		$op_sigil = $this->is_tag_closer() ? '-' : '+';
 		$op       = "{$op_sigil}{$tag_name}";
 
+		if ( null === $tag_name && '#text' === $this->get_token_type() ) {
+			/*
+			 * This rule is necessary even without supporting text nodes in the
+			 * HTML Processor because the parser has to move past text nodes, and
+			 * there could be breadcrumb implications when the text triggers the
+			 * active format reconstruction.
+			 *
+			 */
+			$this->reconstruct_active_formatting_elements();
+
+			$current_token = $this->bookmarks[ $this->state->current_token->bookmark_name ];
+			if ( 1 === $current_token->length && "\x00" === $this->html[ $current_token->start ] ) {
```

Comment:
this needs to check instead if the full content of the text node is null bytes, not only if it's a single null byte